### ISSUE TITLE
[BUG] SOUND file will not be imported untill you manually add one frame to the SoundLayer

### ISSUE THREAD

**Issue by Kaikogf** [pencil2d/pencil#768]
### --Issue Summary--
Sound file will not be added successfully to its Sound Layer until there is at least one frame

### --Actual Results
Importing a sound file will not be correctly imported until a frame is present on its Sound Layer

### --Expected Results--
Ready to playback the sound file as soon as it is imported

### --Steps to reproduce--
1. Open Pencil2D as new.
2. Go to FILE > IMPORT > SOUND. A pop-up window will ask to "create sound layer". Create one and then select the sound file.
4. Pencil2D will show a **Sound Layer** on the time line. When hitting the playing button there is no sound. 
[note: as an attempt to verify if the sound file was initially imported correctly by Pencil2D, I went again to File >Import>Sound to pick up another file but  a window "A sound clip already exists on this frame!..."  shows up, meaning there is a sound file already imported but Pencil2D is unable to reproduce it since there is no sound at playback]

5. Now go to the **Sound Layer** and ADD a NEW FRAME [plus signal on the timeline]: RESULT: A  pop-up window will open asking to chose the sound file and it will be imported and now Pencil2D will be able to play it with sound.

[ANOTHER ISSUE, I rather to have it in just one post:
when adding a new sound layer, and importing a sound file and playing back with it and then decide to close Pencil2D, it will not ask for saving the actual changes and will close immediately. I think this can be preventing by showing the pop-up warning window]

### --System Information--
+ Pencil2D Version:

+ Operating System:
WIND 8.1x64bits

+ RAM Size:
32GB

+ Graphics Tablet:
HUION GT185 HD

**Comment by CandyFace** [comment]
I can't seem to reproduce the first issue, although i've had some weird problems with the timeline too when importing sounds, they've always been periodic and not reproducible.

Your second issue is valid, although that's the intentional behaviour atm. because only drawing on the canvas will trigger such message. Creating frames and interacting with the timeline does not count.

**Comment by Kaikogf** [comment]
[I edited the steps to recreate the first Issue so it could be easily understood. I reproduced it again following the steps above so it is a consistent Issue though.
The second issue I found it to be as such because I was to believe that any change in Pencil2D would have triggered the warning window]

**Comment by chchwy** [comment]
I couldn't replicate issue 1 either.
I guess maybe it is related to some specific audio files? Could you share your audio file with us?

**Comment by Jose-Moreno** [comment]
@Kaikogf @CandyFace @chchwy 

I think there's some information that you need to provide such as Nightly Build version and the filetype for the audio sample you are using.

As for me, I am using Pencil2D September 23rd 2017 build, on Windows 7, I can't reproduce the initial issue. I take the same steps you outlined (and edited), and I can import the file and play it normally on the timeline. 

I too get the following warning when trying to import any other sound to the same "frame"

![image](https://user-images.githubusercontent.com/4912907/31917392-f12e37f4-b81c-11e7-9127-a2f2f68e4498.png)

But this is just a precaution so you don't inadvertedly overwrite your sounds by mistake.

Please have in mind that you can only import WAV and MP3 files, any other file is not currently supported like OGG or AAC (which in due time we should be able to)

Also, even if you are importing a file that displays as either WAV or MP3, be sure to check it's enconding information, sometimes if WAV files are not properly signed as 16bit audio files, they will not work (just had this issue with Davinci Resolve last week), you should be able to check this information with Audacity, VLCPlayer or even with a video Editor like KDEnlive

The second issue, I agree this should be a feature request because importing media should trigger the save flag so Pencil2D can warn the user to save their work. even if they just imported the sound file, although normally one would at least draw or delete a drawing before trying to quit hehe.

**Comment by Kaikogf** [comment]
Hi all

I was waiting for new changes to the Pencil2D code here in order to check if this Issue was still present.
I am now testing the latest ZIP file's changes (28OCT2017) and I can confirm that this Issue is no longer there. So I will close this post and add another post for "Request" changes on the sound file to be followed by a warning pop-up window before closing Pencil2D

I have tested it with **_.wav_** and **_mp3_** files and it worked as expected (those where the same files I used before when I firstly noticed this Issue) 👍 

@Jose-Moreno 
Not sure how to check the sound file information. Even changing its extension I end up with VLCPlayer telling me that the sound file is actually a .wav extension (for instance) which was not initially. Or checking to see the 16bit information 👍
